lib/nerve/reporter/zookeeper.rb

```diff
@@ -26,7 +28,7 @@ def initialize(service)
       @zk_cluster = host_list_to_cluster(zk_host_list)
       @zk_connection_string = zk_host_list.join(',')
       @data = parse_data(get_service_data(service))
-
+      @mode = (service['node_type'] || DEFAULT_NODE_TYPE).to_sym
```

Comment:
One last thought: add some validation of this option (that it is an allowed type) & unit tests for that validation

Reply:
Ruby ZK client actually handles this already https://github.com/zk-ruby/zk/blob/155db85653c3831cc005140b5fd58adc80a54d8e/lib/zk/client/base.rb#L393. So prefer to keep it clean here. Also, it is an option set by the user so that users should realize if they set something wrong.